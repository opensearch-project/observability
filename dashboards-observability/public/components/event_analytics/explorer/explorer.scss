/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

#opensearch-dashboards-body {
  overflow-y: hidden;
}

.liveStream {
  margin : 8px;
  height: 40px;
  align-items: center;
  justify-content: center;
  flex-direction: row;
  display: flex;
  flex-grow: 1;
  vertical-align: baseline;
}

.mainContentTabs .euiResizableContainer {
  height: calc(100vh - 298px);
}
  