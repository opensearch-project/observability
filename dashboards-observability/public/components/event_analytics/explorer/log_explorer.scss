/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

.queryTabs {
  .euiTabs {
    .tab-title {
      max-width: 8rem;
      overflow: hidden;
      white-space: nowrap;
      text-overflow: ellipsis;
    }
    svg {
      vertical-align: inherit;
    }
    .linkNewTag{
      display: inline-block;
      text-align: center;
      font-size: 0.875rem;
      line-height: 3.5;
      padding: 0.5rem;
      min-width: 6rem;
    }
  }
}

.tab-title {
  display: inline-block;
  margin-right: 5px;
}

.search-area {
  position: relative;
}