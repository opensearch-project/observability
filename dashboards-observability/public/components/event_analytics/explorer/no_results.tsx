/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { FormattedMessage, I18nProvider } from '@osd/i18n/react';
import { EuiCallOut, EuiFlexGroup, EuiFlexItem, EuiSpacer, EuiText } from '@elastic/eui';

export const NoResults = () => {
  return (
    <I18nProvider>
      <>
        <EuiSpacer size="xl" />

        <EuiFlexGroup justifyContent="center">
          <EuiFlexItem grow={false} className="dscNoResults">
            <EuiCallOut
              title={
                <FormattedMessage
                  id="discover.noResults.searchExamples.noResultsMatchSearchCriteriaTitle"
                  defaultMessage="No results match your search criteria"
                />
              }
              color="warning"
              iconType="help"
              data-test-subj="discoverNoResults"
            />
            <>
              <EuiSpacer size="xl" />
              <EuiText>
                <h2 data-test-subj="discoverNoResultsTimefilter">
                  <FormattedMessage
                    id="discover.noResults.expandYourTimeRangeTitle"
                    defaultMessage="Expand your time range or modify your query"
                  />
                </h2>
                <p>
                  <FormattedMessage
                    id="discover.noResults.queryMayNotMatchTitle"
                    defaultMessage="Your query may not match anything in the current time range, or there may not be any data at all in the currently selected time range. Try change time range, query filters or choose different time fields"
                  />
                </p>
              </EuiText>
            </>
          </EuiFlexItem>
        </EuiFlexGroup>
      </>
    </I18nProvider>
  );
};
