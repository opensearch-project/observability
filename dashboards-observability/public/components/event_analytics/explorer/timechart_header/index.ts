/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export { TimechartHeader } from './timechart_header';
