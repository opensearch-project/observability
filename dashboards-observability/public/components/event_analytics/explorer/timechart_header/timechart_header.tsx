/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { EuiFlexGroup, EuiFlexItem, EuiToolTip, EuiText, EuiSelect } from '@elastic/eui';
import { I18nProvider } from '@osd/i18n/react';
import { i18n } from '@osd/i18n';

export interface TimechartHeaderProps {
  /**
   * Format of date to be displayed
   */
  dateFormat?: string;
  /**
   * Range of dates to be displayed
   */
  timeRange?: {
    from: string;
    to: string;
  };
  /**
   * Interval Options
   */
  options: Array<{ text: string; value: string }>;
  /**
   * changes the interval
   */
  onChangeInterval: (interval: string) => void;
  /**
   * selected interval
   */
  stateInterval?: string | undefined;
}

export function TimechartHeader({
  options,
  onChangeInterval,
  stateInterval,
}: TimechartHeaderProps) {

  const handleIntervalChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
    onChangeInterval(e.target.value);
  };

  return (
    <I18nProvider>
      <EuiFlexGroup gutterSize="s" responsive justifyContent="center" alignItems="center">
        <EuiFlexItem grow={false}>
          <EuiToolTip
            content={i18n.translate('discover.howToChangeTheTimeTooltip', {
              defaultMessage: 'To change the time, use the global time filter above',
            })}
            delay="long"
          >
            <EuiText data-test-subj="discoverIntervalDateRange" size="s"></EuiText>
          </EuiToolTip>
        </EuiFlexItem>
        <EuiFlexItem grow={false}>
          <EuiSelect
            aria-label={i18n.translate('discover.timechartHeader.timeIntervalSelect.ariaLabel', {
              defaultMessage: 'Time interval',
            })}
            compressed
            id="dscResultsIntervalSelector"
            data-test-subj="eventAnalytics__EventIntervalSelect"
            options={options}
            value={stateInterval || options[0].value}
            onChange={handleIntervalChange}
            append={undefined}
          />
        </EuiFlexItem>
      </EuiFlexGroup>
    </I18nProvider>
  );
}
