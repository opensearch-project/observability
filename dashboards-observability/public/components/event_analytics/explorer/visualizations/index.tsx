/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import './app.scss';

import { isEmpty } from 'lodash';

import React from 'react';
import { EuiResizableContainer } from '@elastic/eui';
import { QueryManager } from 'common/query_manager';
import { RAW_QUERY, SELECTED_TIMESTAMP } from '../../../../../common/constants/explorer';
import {
  IField,
  IQuery,
  IVisualizationContainerProps,
  ExplorerFields,
} from '../../../../../common/types/explorer';
import { WorkspacePanel } from './workspace_panel';
import { ConfigPanel } from './config_panel';
import { Sidebar } from '../sidebar';
import { DataConfigPanelItem } from './config_panel/config_panes/config_controls/data_configurations_panel';
import { PPL_STATS_REGEX, VIS_CHART_TYPES } from '../../../../../common/constants/shared';
import { TreemapConfigPanelItem } from './config_panel/config_panes/config_controls/treemap_config_panel_item';
import { LogsViewConfigPanelItem } from './config_panel/config_panes/config_controls/logs_view_config_panel_item';

interface IExplorerVisualizationsProps {
  query: IQuery;
  curVisId: string;
  setCurVisId: (visId: string) => void;
  explorerVis: any;
  explorerFields: ExplorerFields;
  explorerData: any;
  handleAddField: (field: IField) => void;
  handleRemoveField: (field: IField) => void;
  visualizations: IVisualizationContainerProps;
  handleOverrideTimestamp: (field: IField) => void;
  callback?: any;
  queryManager: QueryManager;
}

export const ExplorerVisualizations = ({
  query,
  curVisId,
  setCurVisId,
  explorerVis,
  explorerFields,
  explorerData,
  handleAddField,
  handleRemoveField,
  visualizations,
  handleOverrideTimestamp,
  callback,
  queryManager,
}: IExplorerVisualizationsProps) => {
  const { vis } = visualizations;
  const fieldOptionList = explorerFields.availableFields.map((field) => ({
    ...field,
    label: field.name,
  }));

  const renderDataConfigContainer = () => {
    switch (curVisId) {
      case VIS_CHART_TYPES.TreeMap:
        return (
          <TreemapConfigPanelItem
            fieldOptionList={fieldOptionList}
            visualizations={visualizations}
          />
        );
      case VIS_CHART_TYPES.LogsView:
        return (
          <LogsViewConfigPanelItem
            fieldOptionList={fieldOptionList}
            visualizations={visualizations}
          />
        );
      default:
        return (
          <DataConfigPanelItem
            fieldOptionList={fieldOptionList}
            visualizations={visualizations}
            queryManager={queryManager}
          />
        );
    }
  };

  const syntheticResize = () => {
    const event = window.document.createEvent('UIEvents');
    event.initUIEvent('resize', true, false, window, 0);
    window.dispatchEvent(event);
  };

  return (
    <div id="vis__mainContent">
      <EuiResizableContainer onPanelWidthChange={syntheticResize}>
        {(EuiResizablePanel, EuiResizableButton) => (
          <>
            <EuiResizablePanel
              initialSize={20}
              minSize="17%"
              mode={['collapsible', { position: 'top' }]}
              paddingSize="none"
              className="vis__leftPanel"
            >
              <div className="explorer__insights">
                <div className="explorerFieldSelector">
                  <Sidebar
                    query={query}
                    explorerFields={explorerFields}
                    explorerData={explorerData}
                    selectedTimestamp={visualizations?.data?.query[SELECTED_TIMESTAMP] || ''}
                    handleOverrideTimestamp={handleOverrideTimestamp}
                    handleAddField={(field: IField) => handleAddField(field)}
                    handleRemoveField={(field: IField) => handleRemoveField(field)}
                    isFieldToggleButtonDisabled={
                      vis.name === VIS_CHART_TYPES.LogsView
                        ? isEmpty(explorerData.jsonData) ||
                          !isEmpty(query[RAW_QUERY].match(PPL_STATS_REGEX))
                        : true
                    }
                  />
                </div>
                <div className="explorer__vizDataConfig">{renderDataConfigContainer()}</div>
              </div>
            </EuiResizablePanel>
            <EuiResizableButton />
            <EuiResizablePanel
              className="ws__central--canvas"
              initialSize={60}
              minSize="55%"
              mode="main"
              paddingSize="none"
            >
              <WorkspacePanel
                curVisId={curVisId}
                setCurVisId={setCurVisId}
                visualizations={visualizations}
              />
            </EuiResizablePanel>
            <EuiResizableButton />
            <EuiResizablePanel
              className="ws__configPanel--right"
              initialSize={20}
              minSize="15%"
              mode={['collapsible', { position: 'top' }]}
              paddingSize="none"
            >
              <ConfigPanel
                vizVectors={explorerVis}
                visualizations={visualizations}
                curVisId={curVisId}
                setCurVisId={setCurVisId}
                callback={callback}
              />
            </EuiResizablePanel>
          </>
        )}
      </EuiResizableContainer>
    </div>
  );
};
