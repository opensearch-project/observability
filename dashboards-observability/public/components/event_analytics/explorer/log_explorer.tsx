/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
/* eslint-disable react-hooks/exhaustive-deps */

import './log_explorer.scss';
import React, { useEffect, useMemo, useRef, useState } from 'react';
import { useDispatch, useSelector, batch } from 'react-redux';
import { uniqueId, map, isEmpty } from 'lodash';
import $ from 'jquery';
import { EuiIcon, EuiText, EuiTabbedContentTab, EuiTabbedContent } from '@elastic/eui';
import { Explorer } from './explorer';
import { ILogExplorerProps } from '../../../../common/types/explorer';
import {
  TAB_TITLE,
  TAB_ID_TXT_PFX,
  SAVED_OBJECT_ID,
  NEW_TAB,
  REDIRECT_TAB,
  TAB_EVENT_ID,
  TAB_CHART_ID,
  APP_ANALYTICS_TAB_ID_REGEX,
} from '../../../../common/constants/explorer';
import { selectQueryTabs, setSelectedQueryTab } from '../redux/slices/query_tab_slice';
import { selectQueries } from '../redux/slices/query_slice';
import { selectQueryResult } from '../redux/slices/query_result_slice';
import { initializeTabData, removeTabData } from '../../application_analytics/helpers/utils';

const searchBarConfigs = {
  [TAB_EVENT_ID]: {
    showSaveButton: true,
    showSavePanelOptionsList: false,
  },
  [TAB_CHART_ID]: {
    showSaveButton: true,
    showSavePanelOptionsList: true,
  },
};

export const LogExplorer = ({
  pplService,
  dslService,
  savedObjects,
  timestampUtils,
  setToast,
  savedObjectId,
  getExistingEmptyTab,
  history,
  notifications,
  http,
  queryManager,
}: ILogExplorerProps) => {
  const dispatch = useDispatch();
  const tabIds = useSelector(selectQueryTabs).queryTabIds.filter(
    (tabid: string) => !tabid.match(APP_ANALYTICS_TAB_ID_REGEX)
  );
  const tabNames = useSelector(selectQueryTabs).tabNames;
  const queries = useSelector(selectQueries);
  const curSelectedTabId = useSelector(selectQueryTabs).selectedQueryTab;
  const explorerData = useSelector(selectQueryResult);
  const queryRef = useRef();
  const tabIdsRef = useRef();
  const explorerDataRef = useRef();
  const curSelectedTabIdRef = useRef();
  queryRef.current = queries;
  tabIdsRef.current = tabIds;
  explorerDataRef.current = explorerData;
  curSelectedTabIdRef.current = curSelectedTabId;

  const [tabCreatedTypes, setTabCreatedTypes] = useState({});

  // Append add-new-tab link to the end of the tab list, and remove it once tabs state changes
  useEffect(() => {
    const newLink = $(
      '<a class="linkNewTag" data-test-subj="eventExplorer__addNewTab">+ Add new</a>'
    ).on('click', () => {
      addNewTab(NEW_TAB);
    });
    $('.queryTabs > .euiTabs').append(newLink);
    return () => {
      $('.queryTabs > .euiTabs .linkNewTag').remove();
    };
  }, [tabIds]);

  const handleTabClick = (selectedTab: EuiTabbedContentTab) => {
    history.replace(
      `/event_analytics/explorer/${queryRef.current![selectedTab.id][SAVED_OBJECT_ID] || ''}`
    );
    dispatch(setSelectedQueryTab({ tabId: selectedTab.id }));
  };

  const handleTabClose = (TabIdToBeClosed: string) => {
    if (tabIds.length === 1) {
      setToast('Cannot close last tab.', 'danger');
      return;
    }

    const index: number = tabIds.indexOf(TabIdToBeClosed);
    const curSelectedTab = curSelectedTabIdRef.current;
    let newIdToFocus = '';
    if (TabIdToBeClosed === curSelectedTab) {
      if (index === 0) {
        newIdToFocus = tabIds[index + 1];
      } else if (index > 0) {
        newIdToFocus = tabIds[index - 1];
      }
    }
    removeTabData(dispatch, TabIdToBeClosed, newIdToFocus);
  };

  const addNewTab = async (where: string) => {
    // get a new tabId
    const tabId = uniqueId(TAB_ID_TXT_PFX);

    // create a new tab
    await initializeTabData(dispatch, tabId, where);

    setTabCreatedTypes((staleState) => {
      return {
        ...staleState,
        [tabId]: where,
      };
    });

    return tabId;
  };

  const dispatchSavedObjectId = async () => {
    const emptyTabId = getExistingEmptyTab({
      tabIds: tabIdsRef.current,
      queries: queryRef.current,
      explorerData: explorerDataRef.current,
    });
    const newTabId = emptyTabId ? emptyTabId : await addNewTab(REDIRECT_TAB);
    return newTabId;
  };

  useEffect(() => {
    if (!isEmpty(savedObjectId)) {
      dispatchSavedObjectId();
    }
  }, []);

  function getQueryTab({
    tabTitle,
    tabId,
    handlesTabClose,
  }: {
    tabTitle: string;
    tabId: string;
    handlesTabClose: (TabIdToBeClosed: string) => void;
  }) {
    return {
      id: tabId,
      name: (
        <>
          <EuiText size="s" textAlign="left" color="default">
            <span className="tab-title">{tabTitle}</span>
            <EuiIcon
              type="cross"
              onClick={(e) => {
                e.stopPropagation();
                handlesTabClose(tabId);
              }}
              data-test-subj="eventExplorer__tabClose"
            />
          </EuiText>
        </>
      ),
      content: (
        <>
          <Explorer
            key={`explorer_${tabId}`}
            pplService={pplService}
            dslService={dslService}
            tabId={tabId}
            savedObjects={savedObjects}
            timestampUtils={timestampUtils}
            setToast={setToast}
            history={history}
            notifications={notifications}
            savedObjectId={savedObjectId}
            tabCreatedTypes={tabCreatedTypes}
            curSelectedTabId={curSelectedTabIdRef}
            http={http}
            searchBarConfigs={searchBarConfigs}
            queryManager={queryManager}
          />
        </>
      ),
    };
  }

  const memorizedTabs = useMemo(() => {
    const res = map(tabIds, (tabId) => {
      return getQueryTab({
        tabTitle: tabNames[tabId] || TAB_TITLE,
        tabId,
        handlesTabClose: handleTabClose,
      });
    });

    return res;
  }, [tabIds, tabNames, tabCreatedTypes]);

  return (
    <>
      <EuiTabbedContent
        id="queryTabs"
        className="queryTabs"
        tabs={memorizedTabs}
        selectedTab={memorizedTabs.find((tab) => tab.id === curSelectedTabId)}
        onTabClick={(selectedTab: EuiTabbedContentTab) => handleTabClick(selectedTab)}
        data-test-subj="eventExplorer__topLevelTabbing"
      />
    </>
  );
};
