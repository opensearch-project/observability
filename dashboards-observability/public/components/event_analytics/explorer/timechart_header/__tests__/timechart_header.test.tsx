/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { waitFor } from '@testing-library/react';
import { TimechartHeader } from '../timechart_header';
import { TIME_INTERVAL_OPTIONS } from '../../../../../../common/constants/explorer';

describe('Time chart header component', () => {
  configure({ adapter: new Adapter() });

  it('Renders Time chart header component', async () => {
    const onChangeInterval = jest.fn();

    const wrapper = mount(
      <TimechartHeader
        dateFormat={'MMM D, YYYY @ HH:mm:ss.SSS'}
        onChangeInterval={onChangeInterval}
        options={TIME_INTERVAL_OPTIONS}
        stateInterval="w"
      />
    );

    wrapper.update();

    await waitFor(() => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
