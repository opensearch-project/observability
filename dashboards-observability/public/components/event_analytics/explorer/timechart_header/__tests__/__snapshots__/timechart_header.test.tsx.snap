// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Time chart header component Renders Time chart header component 1`] = `
<TimechartHeader
  dateFormat="MMM D, YYYY @ HH:mm:ss.SSS"
  onChangeInterval={[MockFunction]}
  options={
    Array [
      Object {
        "text": "Minute",
        "value": "m",
      },
      Object {
        "text": "Hour",
        "value": "h",
      },
      Object {
        "text": "Day",
        "value": "d",
      },
      Object {
        "text": "Week",
        "value": "w",
      },
      Object {
        "text": "Month",
        "value": "M",
      },
      Object {
        "text": "Year",
        "value": "y",
      },
    ]
  }
  stateInterval="w"
>
  <I18nProvider>
    <IntlProvider
      defaultLocale="en"
      formats={
        Object {
          "date": Object {
            "full": Object {
              "day": "numeric",
              "month": "long",
              "weekday": "long",
              "year": "numeric",
            },
            "long": Object {
              "day": "numeric",
              "month": "long",
              "year": "numeric",
            },
            "medium": Object {
              "day": "numeric",
              "month": "short",
              "year": "numeric",
            },
            "short": Object {
              "day": "numeric",
              "month": "numeric",
              "year": "2-digit",
            },
          },
          "number": Object {
            "currency": Object {
              "style": "currency",
            },
            "percent": Object {
              "style": "percent",
            },
          },
          "relative": Object {
            "days": Object {
              "units": "day",
            },
            "hours": Object {
              "units": "hour",
            },
            "minutes": Object {
              "units": "minute",
            },
            "months": Object {
              "units": "month",
            },
            "seconds": Object {
              "units": "second",
            },
            "years": Object {
              "units": "year",
            },
          },
          "time": Object {
            "full": Object {
              "hour": "numeric",
              "minute": "numeric",
              "second": "numeric",
              "timeZoneName": "short",
            },
            "long": Object {
              "hour": "numeric",
              "minute": "numeric",
              "second": "numeric",
              "timeZoneName": "short",
            },
            "medium": Object {
              "hour": "numeric",
              "minute": "numeric",
              "second": "numeric",
            },
            "short": Object {
              "hour": "numeric",
              "minute": "numeric",
            },
          },
        }
      }
      locale="en"
      messages={Object {}}
      textComponent={Symbol(react.fragment)}
    >
      <PseudoLocaleWrapper>
        <EuiFlexGroup
          alignItems="center"
          gutterSize="s"
          justifyContent="center"
          responsive={true}
        >
          <div
            className="euiFlexGroup euiFlexGroup--gutterSmall euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentCenter euiFlexGroup--directionRow euiFlexGroup--responsive"
          >
            <EuiFlexItem
              grow={false}
            >
              <div
                className="euiFlexItem euiFlexItem--flexGrowZero"
              >
                <EuiToolTip
                  content="To change the time, use the global time filter above"
                  delay="long"
                  position="top"
                >
                  <span
                    className="euiToolTipAnchor"
                    onKeyUp={[Function]}
                    onMouseOut={[Function]}
                    onMouseOver={[Function]}
                  >
                    <EuiText
                      data-test-subj="discoverIntervalDateRange"
                      onBlur={[Function]}
                      onFocus={[Function]}
                      size="s"
                    >
                      <div
                        className="euiText euiText--small"
                        data-test-subj="discoverIntervalDateRange"
                        onBlur={[Function]}
                        onFocus={[Function]}
                      />
                    </EuiText>
                  </span>
                </EuiToolTip>
              </div>
            </EuiFlexItem>
            <EuiFlexItem
              grow={false}
            >
              <div
                className="euiFlexItem euiFlexItem--flexGrowZero"
              >
                <EuiSelect
                  aria-label="Time interval"
                  compressed={true}
                  data-test-subj="eventAnalytics__EventIntervalSelect"
                  id="dscResultsIntervalSelector"
                  onChange={[Function]}
                  options={
                    Array [
                      Object {
                        "text": "Minute",
                        "value": "m",
                      },
                      Object {
                        "text": "Hour",
                        "value": "h",
                      },
                      Object {
                        "text": "Day",
                        "value": "d",
                      },
                      Object {
                        "text": "Week",
                        "value": "w",
                      },
                      Object {
                        "text": "Month",
                        "value": "M",
                      },
                      Object {
                        "text": "Year",
                        "value": "y",
                      },
                    ]
                  }
                  value="w"
                >
                  <EuiFormControlLayout
                    compressed={true}
                    fullWidth={false}
                    icon={
                      Object {
                        "side": "right",
                        "type": "arrowDown",
                      }
                    }
                    inputId="dscResultsIntervalSelector"
                    isLoading={false}
                  >
                    <div
                      className="euiFormControlLayout euiFormControlLayout--compressed"
                    >
                      <div
                        className="euiFormControlLayout__childrenWrapper"
                      >
                        <EuiValidatableControl>
                          <select
                            aria-label="Time interval"
                            className="euiSelect euiSelect--compressed"
                            data-test-subj="eventAnalytics__EventIntervalSelect"
                            id="dscResultsIntervalSelector"
                            onChange={[Function]}
                            onMouseUp={[Function]}
                            value="w"
                          >
                            <option
                              key="0"
                              value="m"
                            >
                              Minute
                            </option>
                            <option
                              key="1"
                              value="h"
                            >
                              Hour
                            </option>
                            <option
                              key="2"
                              value="d"
                            >
                              Day
                            </option>
                            <option
                              key="3"
                              value="w"
                            >
                              Week
                            </option>
                            <option
                              key="4"
                              value="M"
                            >
                              Month
                            </option>
                            <option
                              key="5"
                              value="y"
                            >
                              Year
                            </option>
                          </select>
                        </EuiValidatableControl>
                        <EuiFormControlLayoutIcons
                          compressed={true}
                          icon={
                            Object {
                              "side": "right",
                              "type": "arrowDown",
                            }
                          }
                          isLoading={false}
                        >
                          <div
                            className="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
                          >
                            <EuiFormControlLayoutCustomIcon
                              size="s"
                              type="arrowDown"
                            >
                              <span
                                className="euiFormControlLayoutCustomIcon"
                              >
                                <EuiIcon
                                  aria-hidden="true"
                                  className="euiFormControlLayoutCustomIcon__icon"
                                  size="s"
                                  type="arrowDown"
                                >
                                  <EuiIconEmpty
                                    aria-hidden={true}
                                    className="euiIcon euiIcon--small euiIcon-isLoading euiFormControlLayoutCustomIcon__icon"
                                    focusable="false"
                                    role="img"
                                    style={null}
                                  >
                                    <svg
                                      aria-hidden={true}
                                      className="euiIcon euiIcon--small euiIcon-isLoading euiFormControlLayoutCustomIcon__icon"
                                      focusable="false"
                                      height={16}
                                      role="img"
                                      style={null}
                                      viewBox="0 0 16 16"
                                      width={16}
                                      xmlns="http://www.w3.org/2000/svg"
                                    />
                                  </EuiIconEmpty>
                                </EuiIcon>
                              </span>
                            </EuiFormControlLayoutCustomIcon>
                          </div>
                        </EuiFormControlLayoutIcons>
                      </div>
                    </div>
                  </EuiFormControlLayout>
                </EuiSelect>
              </div>
            </EuiFlexItem>
          </div>
        </EuiFlexGroup>
      </PseudoLocaleWrapper>
    </IntlProvider>
  </I18nProvider>
</TimechartHeader>
`;
